src/components/pagination/Pagination.tsx

```diff
@@ -1,16 +1,24 @@
 import { type ReactNode, useEffect } from 'react';
-import { useSearchParams } from 'react-router-dom';
+import { useSearchParams, useLocation, useNavigate } from 'react-router-dom';
 import classNames from 'classnames';
+import { useDispatch, useSelector } from 'react-redux';
 import styles from './styles.module.scss';
-import type { ICardProps } from './types.ts';
+import { setPage } from '../../store/slices/paginationSlice';
+import type { RootState } from '../../store/rootReducer.ts';
+import type { IPaginationProps } from './types';
 
 export function Pagination({
```

Comment:
You have the best pagination among my mentees. That's cool.
But here is the problem, somehow related to number 7.
I think it's connected with rounding.
Not critical, but you can easely fix it for now by removing the addition of '...' fragments and get back to it when you have more free time.

![image](https://github.com/user-attachments/assets/5c6a366e-45d3-4c43-83f8-e35710f738ba)